### ISSUE TITLE
[New Rule] Reduce conditional expressions to boolean expressions

### ISSUE THREAD

**Issue by SpecLad** [MartinThoma/flake8-simplify#27]
## Explanation

Sometimes conditional expressions can be replaced with boolean ones, like this:

* `a if a else b` -> `a or b`
* `b if a else a` -> `a and b`

This makes the code shorter and removes redundant evaluations of expressions. I think a check to suggest such replacements would fit well in this plugin.

## Example

```python
# Bad
x = o.x if o.x else default_x

# Good
x = o.x or default_x
```

**Comment by MartinThoma** [comment]
This would be pretty easy to check and implement:

```
IfExp(
                test=Name(id='a', ctx=Load()),
                body=Name(id='a', ctx=Load()),
                orelse=Name(id='b', ctx=Load()),
            ),
```

However, I'm uncertain of I would prefer the left or the right expression. Give me a couple of days to think about it. The right expression is shorter, but the left expression seems clearer to me.

**Comment by MartinThoma** [comment]
The second one I would rather convert to

`b if a else a -> b or a`. It might need less evaluation.

**Comment by SpecLad** [comment]
`b if a else a` is not equivalent to `b or a`.

**Comment by MartinThoma** [comment]
Table for `a if a else b` and `a or b`:

| a   | b   | Result |
| --- | --- | ------ |
| "a" | ""  | "a"    |
| ""  | "b" | "b"    |
| "a" | "b" | "a"    |
| ""  | 0   | 0      |

Table for `b if a else a` and `a and b`:

| a   | b   | Result |
| --- | --- | ------ |
| "a" | ""  | ""     |
| ""  | "b" | ""     |
| "a" | "b" | "b"    |
| ""  | 0   | ""     |

**Comment by MartinThoma** [comment]
@SpecLad You're completely right. They don't even have a single element in common :sweat_smile: 

This example helped me to make my mind up about this rule. Although it easy to reason which input gives which output, I do need to think about it. I find the boolean expressions `a or b` / `a and b` / `b or a` / `b and a` harder to reason about than their ternary expression equivalents.

**Comment by MartinThoma** [comment]
As the ternary expressions (a if a else b) are easier than the logical expressions (a or b), I will not implement this rule.
